flink/src/main/java/org/apache/iceberg/flink/connector/sink/RecordPartitioner.java

```diff
@@ -110,7 +110,7 @@ public void partition(Record record) {
     if (isOptional) {
       // the wrapped position handles null layers
       return new WrappedPositionAccessor(p, accessor);
-    } else if (accessor instanceof PositionAccessor) {
+    } else if (accessor != null && accessor.getClass() == PositionAccessor.class) {
```

Comment:
This is a temporary fix. Please check https://github.com/apache/incubator-iceberg/issues/994 or its PR.